server/bin/pbench-index.py

```diff
@@ -170,9 +181,9 @@ def main(options, name):
 
     idxctx.logger.debug("{}.{}: starting", name, idxctx.TS)
 
-    idx = Index(name, options, idxctx, INCOMING_rp)
+    idx = Index(name, options, idxctx, INCOMING_rp, ARCHIVE_rp, qdir)
 
-    status, tarballs = idx.collect_tb(ARCHIVE_rp, qdir)
+    status, tarballs = idx.collect_tb()
```

Comment:
Reassigning `tarballs` here may not do what you hope (see the "Note" at the end of [this section](https://docs.python.org/3/reference/compound_stmts.html#the-for-statement)).

If you really want to implement this functionality, you should probably use a "queue" model, where you start with a list of tarballs and destroy the list as you process it (e.g., use a `while tarballs` instead of the `for`, and remove the tarball from the list before processing it).

Reply:
yeah, I have thought about the queue earlier but there were different reasons for not using that here. I will look into it again.